File: easy/merged-back-and-forth.js
function combine(arr1, arr2) {
    let finalArr = [];
    let maxLength = Math.max(arr1.length, arr2.length);
  
    for (let i = 0; i < maxLength; i++) {
      if (i < arr1.length) {
        finalArr.push(arr1[i]);
      }
      if (i < arr2.length) {
        finalArr.push(arr2[i]);
      }
    }
  
    return finalArr;
  }
  
  console.log(combine(["a", "b", "c", "d", "e", "f", "g"], [1, 2, 3, 4, 5]));
  

File: Python-leet/merged-back-and-forth.py

def combine(arr1, arr2):
  final_arr = [];
  max_length = max(len(arr1), len(arr2));

  for i in range(max_length):
    if i < len(arr1):
      final_arr.append(arr1[i]);
    if i < len(arr2):
      final_arr.append(arr2[i]);
    
  return final_arr;


print(combine(["a", "b", "c", "d", "e", "f", "g"], [1, 2, 3, 4, 5]));
